packages/peregrine/lib/talons/CartPage/ProductListing/EditModal/useEditModal.js

```diff
@@ -1,11 +1,17 @@
+import { useState } from 'react';
+
 import { useAppContext } from '../../../../context/app';
 
 export const useEditModal = () => {
     const [{ drawer }, { closeDrawer }] = useAppContext();
     const isOpen = drawer === 'product.edit';
 
+    const [variantPrice, setVariantPrice] = useState();
```

Comment:
This is `undefined` initially. What happens if the item in the cart is a configurable item? I know it falls back to `price` but could that be wrong? Perhaps we need to do the calculation here.

Reply:
The first render for this component will have `item = undefined`, so it wouldn't be that easy. I could add an effect, but I didn't really see the purpose when `ProductDetail` already had the data it needed to render price. Instead of having to handle a bunch of different states for opening, then resetting on close, I passed this responsibility down to `ProductForm`.

BTW - you can only edit Configurable product right now (assuming you meant _isn't a configurable item_)

Reply:
Ah yes, I mistyped.

What I meant was that when you first click on a product to edit the value for variant was "undefined". It seemed to render correctly regardless, but maybe that was by coincidence?

Reply:
By [design](https://github.com/magento/pwa-studio/pull/2483/files#diff-75bfab0706e8e99bd7c97dbd3548145dR19). I'm willing to make this change, but did want to do some validation. If I add a product to my cart that is on sale, but don't checkout until a day later when that product is no longer on sale, do I still get that product at the sale price? I don't know the answer to that, but that would be a reason to use cart price instead of variant price. There might be other use cases as well where cart price is the accurate value, and is what we want to render.

Reply:
If variant price reflects product specific sales I think it's fine to only use that. Would be odd to reflect cart sales such as BOGO or <$X discounts in an edit view.

Reply:
> If I add a product to my cart that is on sale, but don't checkout until a day later when that product is no longer on sale, do I still get that product at the sale price? I don't know the answer to that, but that would be a reason to use cart price instead of variant price.

We can assume that if a sale price expires before a shopper completes a purchase, the sale price is forfeit. If there are special cases where that's not true, someone would have to handle those cases as exceptions.